controllers/cloud.redhat.com/providers/iqe/impl.go

```diff
@@ -167,7 +178,55 @@ func CreateIqeJobResource(cache *rc.ObjectCache, cji *crd.ClowdJobInvocation, en
 		logger.Info("No config mounted to the iqe pod")
 	}
 
-	j.Spec.Template.Spec.Containers = []core.Container{c}
+	return nil
+}
+
+// CreateIqeJobResource will create a Job that contains a pod spec for running IQE
+func CreateIqeJobResource(cache *rc.ObjectCache, cji *crd.ClowdJobInvocation, env *crd.ClowdEnvironment, app *crd.ClowdApp, nn types.NamespacedName, ctx context.Context, j *batchv1.Job, logger logr.Logger, client client.Client) error {
```

Comment:
Why not make these methods? It could simplify the prototype considerably I think.

Reply:
Are you referring to all the functions here in this file or just some pieces of CreateIqeJobResource?

And I'm a little confused on what would be the receiver on the method? (I didn't write this originally so was keeping the structure of the file more or less similar other than splitting up this bug function into a couple smaller ones)

Reply:
We chatted on slack and agreed to backlog this refactor backlog that refactor since jobs and CJI's don't currently follow the model of registering a provider